driver/apdu/at.c

```diff
@@ -55,7 +55,7 @@ static int apdu_interface_connect(struct euicc_ctx *ctx)
         device = "/dev/ttyUSB0";
     }
 
-    fuart = fopen(device, "r+");
```

Comment:
It's strange. The "b" flag is a no-op on the POSIX system. It only affects Windows and lets it disable special handling of '\n' and '\x1A'.

Reply:
Although I am unfamiliar with inner workings of the fopen/fread family of functions, this solved the communication issues in my case on Linux. Might've been just luck or some other side effect fixing it. Feel free to close this PR if you don't think this is the real fix.

Reply:
The "b" flag is a no-op on the POSIX system because POSIX has no special handling of '\n' and '\x1A' like Windows. It always behaves like Windows with a "b" flag.
https://github.com/bminor/glibc/blob/c2567b18fd77da9d4a270b20232f12616763bf76/libio/fileops.c#L262-L264
Adding `setbuf(fuart, NULL)` after `fopen()` should fix this problem. You can try it and update the PR.

Reply:
@CoelacanthusHex close the pr?

Reply:
After some more testing, on another device neither rb+ or setbuf fixed this PR.

Closing, since solution was caused by some external factor / not reliably reproducible.